crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -162,33 +172,63 @@ fn bidirectional_channel<T>(buffer: usize) -> [BidirectionalChannelHalf<T>; 2] {
     let (tx_first, rx_first) = mpsc::channel(buffer);
     let (tx_second, rx_second) = mpsc::channel(buffer);
     [
-        BidirectionalChannelHalf {
-            tx: tx_first,
-            rx: rx_second,
-        },
-        BidirectionalChannelHalf {
-            tx: tx_second,
-            rx: rx_first,
-        },
+        BidirectionalChannelHalf::new(tx_first, rx_second),
+        BidirectionalChannelHalf::new(tx_second, rx_first),
     ]
 }
 
 struct BidirectionalChannelHalf<T> {
```

Comment:
Following looking at the way we now send, it feels like `BidirectionalChannelHalf` is a bit too generic. If you used it in the obvious way, just calling `send` and `recv`, it wouldn't do anything. It could do with renaming to reflect this change.